src/JuMP.jl

```diff
@@ -427,11 +427,18 @@ end
 
 index(cr::ConstraintRef) = cr.index
 
-# TODO: Why does this method depend on the type of the constraint?
-# TODO: docstring
-function has_dual(model::Model,
-                         REF::Type{<:ConstraintRef{Model, T}}) where {T <: MOICON}
-    MOI.get(model, MOI.DualStatus()) != MOI.NoSolution
+"""
+    has_dual(con_ref::ConstraintRef)
+
+Return true if the solver has a dual solution available to query, otherwise
+return false.
+
+See also [`dual`](@ref) and [`shadow_price`](@ref).
+"""
+function has_dual(con_ref::ConstraintRef{Model, <:MOICON})
```

Comment:
Since `DualStatus()` is a property of the model and not the ConstraintRef, shouldn't this be `has_dual(::Model)`? That also makes it less misleading to say that the model has duals when potentially not all constraints do.

Consider also renaming to `has_duals` if it becomes a property of the model.

Reply:
Then if latter we support duals for some constraint but not all in MOI we might use `has_dual` for the constraint-wise function and `has_duals` will still be for the model.
`has_dual(::ConstraintRef)` made sense when we had `canget` but now it might be wrong because if some constraints do not have dual while the `DualStatus` is not `NoSolution`, the solver might error so we don't want to give constraint-wise guarantee.